backend/common/src/main/java/ai/verta/modeldb/common/configuration/AppContext.java

```diff
@@ -0,0 +1,52 @@
+package ai.verta.modeldb.common.configuration;
+
+import org.springframework.beans.factory.config.AutowireCapableBeanFactory;
+import org.springframework.beans.factory.support.BeanDefinitionRegistry;
+import org.springframework.beans.factory.support.GenericBeanDefinition;
+import org.springframework.boot.SpringApplication;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.lang.NonNull;
+
+@Configuration
+public class AppContext implements ApplicationContextAware {
+
+  private static AppContext appContext;
+  public ApplicationContext applicationContext;
+
+  public static AppContext getInstance() {
+    if (appContext == null) {
+      appContext = new AppContext();
```

Comment:
might as well just initialize this in the declaration, I think.

Reply:
Resolved